### PR TITLE
Convert decimal points to degrees, minutes and seconds

### LINKED ISSUE #11: Coordinate Conversion
Use conversion libraries and functions such as convert and more

### PR DESCRIPTION
**Pull request by chananaimen**
A function that receives a decimal point and by calculating converts to degrees, minutes and seconds and returns the result.

Closes #11

### CODE REVIEW

**Comment by bdsoha** [comment]
🎉 The first PR to have proper naming conventions based on the assignment.

**Comment by bdsoha** [comment]
### Cleanup
- [ ] Move `myconvertdd` ➡ `coordinate-conversion`
- [ ] Add a `README.md` to `myconvertdd` describing how to install/use your program.

Once complete, we will move on to the code review.

**Comment by chananaimen** [comment]
i dont understand what do you meen : Move myconvertdd ➡ coordinate-conversion? i happy to explantion

**Comment by bdsoha** [comment]
> i dont understand what do you meen : Move myconvertdd ➡ coordinate-conversion? i happy to explantion

Rename the folder to something more significant.

**Comment by chananaimen** [comment]
Call to the folder ' coordinate conversion' ?

**Comment by bdsoha** [comment]
> Call to the folder ' coordinate conversion' ?

Which seems more explanatory to you?
Yes name the folder `coordinate-conversion`.

**Comment by chananaimen** [comment]
נסיתי לעשות לפי הchat gpt:
 אבל אני לא מוצאת שום כפתור לשינוי שם תיקיה
To rename a folder within your fork of a repository on GitHub, you can follow these steps:

    Navigate to your forked repository on GitHub.
    Click on the folder you want to rename to open it.
    Click the "Settings" button in the top-right corner of the folder view.
    Click the "Rename folder" button in the dropdown menu.
    Enter the new name for the folder in the input field and click the "Rename" button.
    GitHub will update the folder name and any references to the folder within your fork of the repository.
אני ממש לא יודעת מה לעשות

**Comment by chananaimen** [comment]
sorry:un englishI tried to do according to the gpt chat but I can't find any button to change the folder name:
To rename a folder within your fork of a repository on GitHub, you can follow these steps:

    Navigate to your forked repository on GitHub.
    Click on the folder you want to rename to open it.
    Click the "Settings" button in the top-right corner of the folder view.
    Click the "Rename folder" button in the dropdown menu.
    Enter the new name for the folder in the input field and click the "Rename" button.
    GitHub will update the folder name and any references to the folder within your fork of the repository.
I really don't know what to do

**Comment by bdsoha** [comment]
Are you working in the web-ui?
You should be working with a local `clone`.

**Comment by chananaimen** [comment]
That's right, I work in local clone new

**Comment by bdsoha** [comment]
Then rename your folder and push the changes

**Comment by chananaimen** [comment]
I made the changes

**Comment by bdsoha** [comment]
`myconvert.js` ❓

**Comment by chananaimen** [comment]
I did not understand.
I changed the folder name to coordinate conversion, should I also change the folder name?
If so, to what name?

**Comment by bdsoha** [comment]
> I did not understand. I changed the folder name to coordinate conversion, should I also change the folder name? If so, to what name?

Based on what I wrote above, you should name files and folders with significant names.
The name `myconvert.js` does not follow that understaning.

**Comment by chananaimen** [comment]
i did it.

**Comment by bdsoha** [comment]
@chananaimen Did you use ChatGPT to generate the code for this assignment?

**Comment by chananaimen** [comment]
No, I made the code myself by searching in google

**Comment by bdsoha** [comment]
Did you run the `docker` container?

**Comment by chananaimen** [comment]
I created a container but now after the changes I try to create a new one but it gives me an error I'm trying to understand why

**Comment by chananaimen** [comment]
should i create a docker?

**Comment by bdsoha** [comment]
> should i create a docker?

Docker images are *immutable*, you need to rebuild the image to apply changes.

**Comment by chananaimen** [comment]
I deleted the previous docker I made and I try to build a new one after the changes but it gives me an error :PS C:\Users\ChanaN\registration\coordinate-conversion> docker build -t convert
"docker build" requires exactly 1 argument.
See 'docker build --help'.

Usage:  docker build [OPTIONS] PATH | URL | -

Build an image from a Dockerfile
שמג I can't find on Google what it means and what to do with it

**Comment by bdsoha** [comment]
You need to provide a *docker context*. You are missing a `.` at the end of your command:

```sh
docker build -t convert .
```

**Comment by chananaimen** [comment]
thank you.
i create a docker
Is the feature that I will do something else

**Comment by bdsoha** [comment]
You are not finished. 
I have yet to do a code review.

File: coordinate-conversion/convertdecimaldegrees.js [hunk 1]
```diff
@@ -0,0 +1,49 @@
+ 
+
+function convertFromDDToDms(dd)
+ {
```
Comment (bdsoha) [inline coordinate-conversion/convertdecimaldegrees.js:4]:
```suggestion
function convertFromDDToDms(dd) {
```

File: coordinate-conversion/convertdecimaldegrees.js [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+     return deg + " degrees " + min + " minutes " + sec + ' second ';
+    }
+     return  deg + " degrees " + min + " minutes " ;
+  }
+  return  deg + " degrees " ;
+ }
+
+console.log(convertFromDDToDms(34.0522));
+ 
+
```
Comment (bdsoha) [inline coordinate-conversion/convertdecimaldegrees.js:49]:
```suggestion
console.log(convertFromDDToDms(34.0522))
```

**Comment by chananaimen** [comment]
Did you mean to have a member of the team go over the code with me?

File: coordinate-conversion/convertdecimaldegrees.js [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+  //   {
+  //     if(isLng)
+  //     {
+  //       dir='E';
+  //     }
+  //     else
+  //     {
+  //       dir='N'
+  //     }
+  //   }
```
Comment (bdsoha) [inline coordinate-conversion/convertdecimaldegrees.js:27]: What is with this code?

File: coordinate-conversion/convertdecimaldegrees.js [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+
+   var absDd = Math.abs(dd);
```
Comment (bdsoha) [inline coordinate-conversion/convertdecimaldegrees.js:29]: Why `var` and not `const`.

File: coordinate-conversion/convertdecimaldegrees.js [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+  {
+     temp=temp*60;
+     var min = Math.floor(temp) ;
+     var sec = temp - min ;
+  if(sec!=0)
+    {
+      sec =(Math.floor(Math.round(sec * 6000)))/100;
+     return deg + " degrees " + min + " minutes " + sec + ' second ';
+    }
+     return  deg + " degrees " + min + " minutes " ;
```
Comment (bdsoha) [inline coordinate-conversion/convertdecimaldegrees.js:42]:
```suggestion
     return  `${deg} degrees ${min} minutes `
```

File: coordinate-conversion/convertdecimaldegrees.js (hunk 4 above)
Comment (chananaimen) [reply coordinate-conversion/convertdecimaldegrees.js]: It is true that it is const because it does not change later in the program.
I made the change

Comment (bdsoha) [reply coordinate-conversion/convertdecimaldegrees.js]: Not only. The `var` keyword is not longer used in *JavaScript* due to its *hoisting* issues.

File: coordinate-conversion/Dockerfile [hunk 6]
```diff
@@ -0,0 +1,7 @@
+FROM node:16.19.1-buster
+
+WORKDIR /coordinate-conversion
```
Comment (bdsoha) [inline coordinate-conversion/Dockerfile:3]: Is there a need for the `WOKRDIR`?

File: coordinate-conversion/README.md [hunk 7]
```diff
@@ -0,0 +1,14 @@
+# Coodinate - conversion
+
+this JavaScript function that called convertFromDDToDms converts decimal degrees to degrees-minutes-seconds in the format "dd degrees mm minutes ss seconds". 
+## Installation
```
Comment (bdsoha) [inline coordinate-conversion/README.md:4]:
```suggestion
This JavaScript function that called convertFromDDToDms converts decimal degrees to degrees-minutes-seconds in the format "dd degrees mm minutes ss seconds".

## Installation
```

File: coordinate-conversion/README.md [hunk 8]
```diff
@@ -0,0 +1,14 @@
...
+
+To use convertFromDDToDms, you need to have a JavaScript environment installed, such as Node.js.
+1. Clone this repository to your local machine.
+2. Open the terminal or command prompt and navigate to the project directory.
+3. Install any dependencies by running the command npm install.
+## Usage
```
Comment (bdsoha) [inline coordinate-conversion/README.md:10]:
```suggestion
3. Install any dependencies by running the command `npm install`.

## Usage
```

File: coordinate-conversion/convertdecimaldegrees.js (hunk 3 above)
Comment (chananaimen) [reply coordinate-conversion/convertdecimaldegrees.js]: I created a function that accepts a decimal point and returns the degrees, minutes and seconds in format "dd degrees mm minutes ss seconds".
It's really unrelated
When I checked I saw various codes that accept a variable called islng and I wrote it down to understand what it was called
And I saw that it was not related to the code, I already deleted it and uploaded it

Comment (bdsoha) [reply coordinate-conversion/convertdecimaldegrees.js]: It is related, and was posted as part of the original assignment.
The earth is split into 4 quarters called hemispheres.
 
image

Comment (chananaimen) [reply coordinate-conversion/convertdecimaldegrees.js]: But what does islang mean?
I did not understand and could not find an explanation

Comment (bdsoha) [reply coordinate-conversion/convertdecimaldegrees.js]: I can assume that `islang` refers to *is langatude*.
One of the reasons you shouldn't be using other peoples code.

**Comment by chananaimen** [comment]
i create a function receives decimal degrees and a boolean variable showing whether it is on a longitude or latitude line and according to this it checks and shows in which direction the point is on the globe

**Comment by bdsoha** [comment]
Using the example provided in the task, will your code convert:

```
[-149.9002, 61.2181, 22] ➡ [[149, 54, 0.72, "W"], [61, 13, 5.16, "N"], 22]
```

**Comment by chananaimen** [comment]
זThis is what the code does. It receives decimal degrees and converts them to degrees minutes and seconds and also receives information whether it is on a longitude or not and accordingly shows in which direction the point

**Comment by bdsoha** [comment]
Please have a look at the other unresolved issues.

**Comment by chananaimen** [comment]
I made changes and I try to push but it gives me an error that I don't know what to do with
error: failed to push some refs to 'https://github.com/chananaimen/registration.git'
hint: Updates were rejected because the remote contains work that you do
hint: not have locally. This is usually caused by another repository pushing
hint: to the same ref. You may want to first integrate the remote changes
hint: (e.g., 'git pull ...') before pushing again.
hint: See the 'Note about fast-forwards' in 'git push --help' for details.

**Comment by bdsoha** [comment]
It says in the error message:

> hint: (e.g., 'git pull ...') before pushing again.

**Comment by bdsoha** [comment]
Did you read about `var` hoisting?

**Comment by chananaimen** [comment]
i dont understand what do you mean?

**Comment by bdsoha** [comment]
> i dont understand what do you mean?

https://github.com/skyvar/registration/pull/27#discussion_r1141326656

**Comment by chananaimen** [comment]
Now I received an explanation why var is no longer used from a company.

**Comment by bdsoha** [comment]
> Now I received an explanation why var is no longer used from a company.

Can you explain that to me in your own words *(shortly)*.

**Comment by chananaimen** [comment]
Because the var is unprotected it can cause errors and the content is also unprotected

**Comment by bdsoha** [comment]
> Because the var is unprotected it can cause errors and the content is also unprotected

Anything regarding the word *scope*?

**Comment by chananaimen** [comment]
lat is recognized only in the scoop and var is recognized on the whole page, so it is not protected.

**Comment by bdsoha** [comment]
Please align and cleanup the code so that it is easier to read.

File: coordinate-conversion/Dockerfile (hunk 6 above)
Comment (bdsoha) [reply coordinate-conversion/Dockerfile:3]: @chananaimen Please answer the question.

File: coordinate-conversion/README.md [hunk 9]
```diff
@@ -0,0 +1,17 @@
+# Coodinate - conversion
+function called convertFromDDToDms converts decimal degrees to degrees-minutes-seconds
+And also shows in which position the degree is located on the globe
+## Installation
```
Comment (bdsoha) [inline coordinate-conversion/README.md:4]:
```suggestion
# Coordinate Conversion
A function called `convertFromDDToDms` converts decimal degrees to degrees-minutes-seconds.
Also shows in which position the degree is located on the globe.

## Installation
```

**Comment by bdsoha** [comment]
🎉 Congrats on your first merged PR.

You will soon be added as a collaborator to this repo.
In the meantime, please prepare a short presentation (in English) describing:

The task in general.
Things you learned along the way.
Tips for other team-members for future tasks.

File: coordinate-conversion/README.md (hunk 9 above)
Comment (chananaimen) [reply coordinate-conversion/README.md]: what do you do?
You deleted everything?

Comment (bdsoha) [reply coordinate-conversion/README.md]: > what do you do? You deleted everything?

I removed a folder with the suffix *עותק*.

Comment (bdsoha) [reply coordinate-conversion/README.md]: Either way, you should never be making backup folders or files. Just utilize `git`'s capabilities to store and stash history.

**Comment by bdsoha** [comment]
You have been sent an email invitation to the [SkyVar Organization](https://github.com/skyvar).
Please open your cooperate email and finish the registration process.

**Pull request merged**